docs/articles_en/openvino-workflow/running-inference/integrate-openvino-with-your-application/model-representation.rst

```diff
@@ -36,6 +36,14 @@ OpenVINO™ Runtime enables you to use different approaches to work with model i
   additional mapping of names from framework to OpenVINO as it was before. OpenVINO Runtime allows the usage of native framework
   tensor names, for example:
 
+.. warning::
+
+   The following example retrives input/output using the name ``index``, which is optional. Some
+   outputs may not have names, so use explicit names and indices if required.
+
+   OpenVINO Runtime has methods to obtain a list of used names (``get_name`` / ``get_any_name``).
+   However, because not all of them have names, using these methods may return an empty list.
```

Comment:
@itikhono, please help to formulate it clearly, give a story. This is not enough, it very confusing formulation.

Reply:
I meant that the ``index`` in the example below is the numeric value (0, 1, 2, ...) that is needed to get the corresponding input or output by its ordinal number (index). This is not name.

Maybe we can re-write this warning this way:

"""
All inputs/outputs of ov::Model are numbered, so the preferred way to retrieve them is to use indices.

Using tensor names can potentially be a less reliable approach, since the mandatory presence of tensor names for inputs and outputs is not guaranteed in the original frameworks. Therefore ov::Model may contain empty list of tensor_names for inputs/outputs.
"""

Reply:
and maybe we can add something like that: 

To get all tensor names which are associated with the corresponding input/output,  OpenVINO Runtime has ``get_names`` method.  To get some name from all names associated with a given input/output, the ``get_any_name`` method was introduced. These methods may return empty names list/empty name if the names are not present.

Reply:
Also, I believe it's better to provide a concrete example for each paragraph here:

* The ``ov::Model::inputs()`` / ``ov::Model::outputs()``  methods are used to get vectors of all input/output ports.

<img width="176" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/76a7ff76-fa07-4b19-8c4d-a1367c088753">

* For a model that has only one input or output, you can use the ``ov::Model::input()`` or ``ov::Model::output()``  methods without
  any arguments to get input or output port respectively.
  
  
  
* The ``ov::Model::input()`` and ``ov::Model::output()``  methods can be used with the index of inputs or outputs from the framework
  model to get specific ports by index.
  
  
<img width="136" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/21043c6f-db5f-425e-928f-0dbdb934a51f">

  
* You can use the tensor name of input or output from the original framework model together with the
  ``ov::Model::input()`` or ``ov::Model::output()`` methods to get specific ports. It means that you do not need to have any
  additional mapping of names from framework to OpenVINO as it was before. OpenVINO Runtime allows the usage of native framework
  tensor names, for example:
  
<img width="191" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/e146c035-a46a-4389-942c-2feb3d0adf55">

Reply:
@itikhono suggestions commited in https://github.com/openvinotoolkit/openvino/pull/23482/commits/a168d84821a71a806e05cdea3eb7d0101fa93d6b